client/src/pages/Campaign.js

```diff
@@ -60,19 +58,33 @@ const Campaign = (props) => {
   const [newOpen, setNewOpen] = useState(false);
   const [confirmOpen, setConfirmOpen] = useState(false);
   const [templateId, setTemplateId] = useState(0);
-  const [emailTemplates, setEmailTemplates] = useState([{}]);
-  const [success, setSuccess] = useState(false);
+  const [templates, setTemplates] = useState([{}]);
+  const [importSuccess, setImportSuccess] = useState(false);
+  const [executeSuccess, setExecuteSuccess] = useState(false);
   const [currentView, setCurrentView] = useState('summary');
   const [selectedProspects, handleSelectedProspects] = useState([]);
   const [prospects, handleProspects] = useState([{}]);
 
   useEffect( () => {
     getCampaign();
-    getEmailTemplates();
+    getTemplates();
     getAllProspects();
   // eslint-disable-next-line react-hooks/exhaustive-deps
   }, []);
 
+  const getSteps = (prospect) => {
+    const listOfSteps = {0: true};
+
+    if (prospect.steps.length === 0) {
+      return listOfSteps;
+    } else {
+      for (let i = 0; i < prospect.steps.length; i += 1) {
+        listOfSteps[prospect.steps[0].id] = true;

```

Comment:
you can use i++,

also shouldn't it be  ``prospect.steps[i].id``

Reply:
prospect.steps looks like this [{id: 8, campaign_id: 5, template_id: 7}]
output: [8]
if prospect has more than 1 steps, then prospect.steps = [{id: 8, campaign_id: 5, template_id: 7}, {id: 11, campaign_id: 6, template_id: 7}]
output: [8, 11]

But I guess, the prospect can only have one step saved for every campaign? so maybe I don't need to loop. I will try refactoring

Reply:
a prospect can be in as many steps and campaigns